### PR TITLE
[java] Deprecate all 'lift style' code

### PR DESCRIPTION
**Pull request by joerg1985**
### Description
As suggested in the tlc Slack channel, this is the proposal to deprecate all 'lift style' related code.
I think this should be removed in Selenium 5 if the community is okay with it. 

### Motivation and Context
- This code has not been updated for one decade, except for refactoring.
- I could not find anything online, only the selenium wiki entry from 2015
- The wiki states: `This is very much work in progress at the moment, ...`
- The code is based on the JUnit4 dependency and the TestCase style is based on JUnit3
- There is an extra jar generated which seens [not to be in use](https://mvnrepository.com/artifact/org.seleniumhq.selenium/lift)

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by titusfortner** [comment]
So yeah, this code is from WebDriver before it merged with Selenium.
Looks like they were created and added to by one person in 2008 & 2009, and have only been moved and had various formatting changes since then.
I don't even think we need to wait for Selenium 5; should be fine with the normal 2 releases then remove. If someone complains we can spin it off. 😆

**Review by diemol — APPROVED** [review]
This makes sense, thank you @joerg1985!

**Pull request merged**
